lib/config.js

```diff
@@ -74,59 +75,102 @@ function validatePolicySection(policy) {
 }
 
 /**
- * Validate configuration (types and values)
+ * Validate code configuration (types and values)
  *
- * @param {Object} opts Options to validate
+ * @param {Object} codes Code section to validate
+ * @param {Number}nLocations Number of available hyperdrives
  * @returns { Object } with decision and error if invalid
  * @returns { boolean } Object.configIsValid
  * @returns { null | InvalidConfigError } Object.ConfigError
  */
-function validate(opts) {
-    if (! (opts instanceof Object)) {
+function validateCodeSection(codes, nLocations) {
+    if (! (codes instanceof Array)) {
         return {
             configIsValid: false,
             configError: new InvalidConfigError(
-                '', '', 'No options passed'),
+                'codes',
+                codes,
+                'Expected an array of { pattern, dataParts, codingParts }'),
         };
     }
 
-    const policyValidated = validatePolicySection(opts.policy);
-    if (!policyValidated.configIsValid) {
-        return policyValidated;
-    }
-
-    if (typeof(opts.dataParts) !== 'number' ||
-        opts.dataParts < 1) {
+    if (codes.length === 0) {
         return {
             configIsValid: false,
             configError: new InvalidConfigError(
-                'dataParts',
-                opts.dataParts,
-                'Expected integer larger than 1'),
+                'codes',
+                codes,
+                'Expected at least one code pattern'),
         };
     }
 
-    if (typeof(opts.codingParts) !== 'number' ||
-        opts.codingParts < 0) {
-        return {
-            configIsValid: false,
-            configError: new InvalidConfigError(
-                'codingParts',
-                opts.codingParts,
-                'Expected integer larger than 0'),
-        };
+    for (let i = 0; i < codes.length; i++) {
+        const pattern = codes[i];
+
+        if (typeof(pattern.dataParts) !== 'number' ||
```

Comment:
C'est moche le JS :(

Reply:
Agreed :( (but perhaps I am not doing it properly either)